Arm/Insts/Common.lean

```diff
@@ -55,6 +55,32 @@ def AddWithCarry (x : BitVec n) (y : BitVec n) (carry_in : BitVec 1) :
   let V := if signExtend (n + 1) result = signed_sum then 0#1 else 1#1
   (result, (make_pstate N Z C V))
 
+/-- When the carry bit is `0`, `AddWithCarry x y 0 = x + y` -/
+theorem fst_AddWithCarry_eq_add (x : BitVec n) (y : BitVec n) :
```

Comment:
Let's also add the `state_simp_rules` tag to these `AddWithCarry` theorems.

Though... I'm getting worried about bloat there. We should consider defining another `simp` attr that can help in simplifying a goal state after symbolic simulation is done.

Reply:
@shigoel same concern :) tagging these as `state_simp_rules` means I need to change `AddTandem`, so I'd rather do that in a separate PR. Does that work for you?